systray.c

```diff
@@ -413,4 +456,29 @@ luaA_systray(lua_State *L)
     return 2;
 }
 
+/** Return the native surface of the systray if composite is enabled.
+ * \param L The Lua VM state.
+ * \return the number of element returned. (1)
+  * \luastack
+ * \lparam width The width of the systray surface.
+ * \lparam height The height of the systray surface.
+ */
+int
+luaA_systray_surface(lua_State *L)
+{
+    if (!globalconf.systray.is_composited) {
+        lua_pushnil(L);
+        return 1;
+    }
+
+    int width = luaL_checkinteger(L, 1);
+    int height = luaL_checkinteger(L, 2);
+    /* Lua has to make sure to free the ref or we have a leak */
+    lua_pushlightuserdata(
+        L, cairo_xcb_surface_create(
+            globalconf.connection, globalconf.systray.window, globalconf.visual,
+            width, height));
```

Comment:
Lua does no memory management on light userdata, so we do have a memory leak.
How would it even know to call `cairo_surface_destroy` on this particular `void*`?

Reply:
Similar to `pixbuf_to_surface`, it will be the caller's duty to manage/destroy the returned resource. I believe that this is done in https://github.com/awesomeWM/awesome/pull/3664/files#diff-8dc3a9fa2a4507278e3c9c437bc5010216606deca220f15f825c3fbef718f06aR109 by calling `cairo.Surface([raw userdata], true)`, but I can be missing something.